File: components/permissions/src/main/java/org/cloudfoundry/credhub/views/PermissionsView.java
package org.cloudfoundry.credhub.views;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import org.apache.commons.lang3.builder.EqualsBuilder;
import org.apache.commons.lang3.builder.HashCodeBuilder;
import org.cloudfoundry.credhub.requests.PermissionEntry;

@JsonAutoDetect
@SuppressWarnings("unused")
public class PermissionsView {

  private String credentialName;
  private List<PermissionEntry> permissions;

  public PermissionsView() {
    super();
  }

  public PermissionsView(final String credentialName,
                         final List<PermissionEntry> permissions) {
    super();
    this.credentialName = credentialName;
    this.permissions = permissions;
  }

  public String getCredentialName() {
    return credentialName;
  }

  public void setCredentialName(final String credentialName) {
    this.credentialName = credentialName;
  }

  public List<PermissionEntry> getPermissions() {
    return permissions;
  }

  public void setPermissions(final List<PermissionEntry> permissions) {
    this.permissions = permissions;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }

    if (o == null || getClass() != o.getClass()) {
      return false;
    }

    PermissionsView that = (PermissionsView) o;

    return new EqualsBuilder()
      .append(credentialName, that.credentialName)
      .append(permissions, that.permissions)
      .isEquals();
  }

  @Override
  public int hashCode() {
    return new HashCodeBuilder(17, 37)
      .append(credentialName)
      .append(permissions)
      .toHashCode();
  }
}

File: components/permissions/src/main/kotlin/org/cloudfoundry/credhub/views/PermissionsView.kt
package org.cloudfoundry.credhub.views

import com.fasterxml.jackson.annotation.JsonAutoDetect
import org.apache.commons.lang3.builder.EqualsBuilder
import org.apache.commons.lang3.builder.HashCodeBuilder
import org.cloudfoundry.credhub.requests.PermissionEntry

@JsonAutoDetect
class PermissionsView {
    var credentialName: String? = null
    var permissions: List<PermissionEntry>? = null

    constructor() : super() {}

    constructor(credentialName: String, permissions: List<PermissionEntry>) : super() {
        this.credentialName = credentialName
        this.permissions = permissions
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }

        if (other == null || javaClass != other.javaClass) {
            return false
        }

        val that = other as PermissionsView?

        return EqualsBuilder()
            .append(credentialName, that!!.credentialName)
            .append(permissions, that.permissions)
            .isEquals
    }

    override fun hashCode(): Int =
        HashCodeBuilder(17, 37)
            .append(credentialName)
            .append(permissions)
            .toHashCode()
}
